### PR TITLE
Get all products by restaurant

### PR DESCRIPTION
**Pull request by RodrigoM2004**

### CODE REVIEW

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+
+class Test_GetAllProductByRestaurant:
+    def test_get_all_product_by_restarant(self):
+        repo = ProductRepositoryMock()
+        usecase = GetAllProductsByRestaurantUsecase(repo=repo)
+
+        product_rest1 = usecase(restaurant=RESTAURANT.SOUZA_DE_ABREU)
+        product_rest2 = usecase(restaurant=RESTAURANT.RESTAURANTE_DO_H)
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: cria 2 testes ao inves de testar os 2 restaurantes em um único teste

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+from src.shared.domain.enums.restaurant_enum import RESTAURANT
+from src.modules.get_all_products_by_restaurant.app.get_all_product_by_restaurant_usecase import GetAllProductsByRestaurantUsecase
+from src.shared.domain.entities.product import Product
+from src.shared.infra.repositories.product_repository_mock import ProductRepositoryMock
+
+
+
+class Test_GetAllProductByRestaurant:
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: Test_GetAllProductByRestaurantUsecase

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    def test_get_all_product_by_restarant(self):
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: test_get_all_product_by_restarant_usecase

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+class ProductViewmodel:
+    product: Product
+
+    def __init__(self, product: Product):
+        self.product = product
+
+    def to_dict(self) -> dict:
+        return{
+            "available": self.product.available,
```
Comment (JoaoVitorBranco) [inline src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py]: identar certo

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+             "restaurant": self.product.restaurant.value,
+             "prepareTime": self.product.prepareTime 
+            }
+
+
+class GetAllProductsByRestaurantViewmodel:
+    all_products: List[ProductViewmodel]
+    
+    def __init__(self, all_products: List[Product]):
+        self.products = [product for product in all_products]
```
Comment (JoaoVitorBranco) [inline src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py]: self.products = all_products

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
+from src.shared.domain.enums.restaurant_enum import RESTAURANT
+from src.modules.get_all_products_by_restaurant.app.get_all_product_by_restaurant_usecase import GetAllProductsByRestaurantUsecase
```
Comment (JoaoVitorBranco) [inline src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py]: from .get_all_product_by_restaurant_usecase import GetAllProductsByRestaurantUsecase

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+from src.modules.get_all_products_by_restaurant.app.get_all_product_by_restaurant_viewmodel import GetAllProductsByRestaurantViewmodel
+from src.shared.helpers.external_interfaces.external_interface import IRequest, IResponse
+from src.shared.helpers.external_interfaces.http_codes import InternalServerError, OK, BadRequest, NotFound
```
Comment (JoaoVitorBranco) [inline src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py]: tira import desnecessário (NotFound por ex)

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        request = HttpRequest(
+            query_params={
+                'restaurant': repo.products[0].restaurant.value
+                }
+            )
+        
+        response = controller(request=request)
+
+        assert response.status_code == 200
+        assert len(response.body['all_products']) == 93
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py:21]: assert all([product.restaurant == repo.products[0].restaurant for product in response.body['all_products'])

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        assert response.body['message'] == "the products were retrieved"
+
+    def test_get_all_product_by_restaurant_controller_missing(self):
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py]: restaurant_is_missing

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+        controller = GetAllProductByRestaurantController(usecase=usecase)
+
+        request = HttpRequest()
+
+        response = controller(request=request)
+
+        assert response.status_code == 400
+        assert response.body == "Field restaurant is missing"
+
+    def test_get_all_product_by_restaurant_controller_EntityError(self):
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py]: invalid_restaurant

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+          "pathParameters": None,
+          "isBase64Encoded": None,
+          "stageVariables": None
+        }
+
+        response = lambda_handler(event, None)
+        
+        assert response["statusCode"] == 200
+    
+    def test_get_all_products_by_restaurant_presenter_missing(self):
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py]: ver nome

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+          },
+          "body": "Visualize all the products",
+          "pathParameters": None,
+          "isBase64Encoded": None,
+          "stageVariables": None
+        }
+
+        response = lambda_handler(event, None)
+        
+        assert response["statusCode"] == 400
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py:1245]: testa mensagem

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+              "timeEpoch": 1583348638390
+            },
+            "body": "Visualize all the products",
+            "pathParameters": None,
+            "isBase64Encoded": None,
+            "stageVariables": None
+          }
+
+          response = lambda_handler(event, None)
+
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py:1303]: testa mensagem

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py [hunk 14]
```diff
@@ -0,0 +1,178 @@
...
+          "body": "Visualize all the products",
+          "pathParameters": None,
+          "isBase64Encoded": None,
+          "stageVariables": None
+        }
+
+        response = lambda_handler(event, None)
+        
+        assert response["statusCode"] == 200
+
```
Comment (JoaoVitorBranco) [inline tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py:1188]: testa o body

**Review by JoaoVitorBranco — CHANGES_REQUESTED** [review]

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py (hunk 1 above)
Comment (RodrigoM2004) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: ok

Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py (hunk 2 above)
Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py (hunk 3 above)
Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_products_by_restaurant_usecase.py]: ok

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py (hunk 4 above)
Comment (LucasKiller) [reply src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py]: ok

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py (hunk 5 above)
Comment (LucasKiller) [reply src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_viewmodel.py]: ok

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py (hunk 6 above)
Comment (LucasKiller) [reply src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py:2]: ok

File: src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py (hunk 7 above)
Comment (LucasKiller) [reply src/modules/get_all_products_by_restaurant/app/get_all_product_by_restaurant_controller.py]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py (hunk 8 above)
Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py:21]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py (hunk 9 above)
Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py (hunk 10 above)
Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_controller.py]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py (hunk 11 above)
Comment (LucasKiller) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py]: ok

File: tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py (hunk 14 above)
Comment (RodrigoM2004) [reply tests/modules/get_all_products_by_restaurant/app/test_get_all_product_by_restaurant_presenter.py:1188]: ok

**Review by JoaoVitorBranco — APPROVED** [review]

**Pull request merged**